sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/encoders/RowEncoderSuite.scala

```diff
@@ -162,6 +162,32 @@ class RowEncoderSuite extends CodegenInterpretedPlanTest {
     assert(row.toSeq(schema).head == decimal)
   }
 
+  test("RowEncoder should respect nullOnOverflow for decimals") {
```

Comment:
nit: may you please also add the Jira number to the test case? Thanks.

Reply:
Updated. thank you Marco!